documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc

```diff
@@ -0,0 +1,128 @@
+[id="ansible-playbook-create-bonded-interfaces-using-script.adoc"]
+
+= Create bonded interfaces
+
+If you need to create a bonded provisioning/baremetal interface, the ifcfg files
+can be placed under the "fake" roots for the masters and workers under the
+`roles/installer/files/customize_filesystem/{master,worker}` directories as shown in the above section 4.7
+
+This can also be done by creating a script /usr/local/bin/create-ifcfg-files.sh under 
+`roles/installer/files/customize_filesystem/{master,worker}`
+and by setting by setting variable create_bond_service="true" to create ifcfg files.
+This file will be included in the ignition config and run as a part of bond.service.
+
+Creation of bonded interfaces using a script has the following flexibility and advantages:
+
+1. The interfaces can be created using one file instead of adding multiple files.
+
+2. This approach will be useful to set DHCP_CLIENT_ID.
+   If we need to set DHCP_CLIENT_ID to help the bonds get a static lease regardless of the MAC Address assigned to it, we could set it to mac address for the interface dynamically.
+   This in conjuction with setting IPV4_DHCP_TIMEOUT to 2147483647 will prevent the bond from getting destroyed if it doesn't get a DHCP lease. 
+   The default value is 45 seconds, however, the value of 2147483647 is equivalent to infinity, meaning the bonding will not be destroyed in case there's no DHCP in the network. 
+   This works as the lease duration is handled by the server,not by the client.
+
+[NOTE]
+====
+Please make sure that the filepath and filename is /usr/local/bin/create-ifcfg-files.sh
+====
+
+An example for creating bonded interfaces bond0 and bond1
+in active backup mode with a vlan interface(VLAN ID 20) on top of bond0 is show below:
+
+[source,bash]
+----
+#!/bin/bash
+
+# bond1 creation
+cat <<EOF > /etc/sysconfig/network-scripts/ifcfg-bond1
+BONDING_OPTS=mode=active-backup
+TYPE=Bond
+BONDING_MASTER=yes
+PROXY_METHOD=none
+ONBOOT=yes
+BOOTPROTO=dhcp
+DHCP_CLIENT_ID=$(ip link show dev ens3f0 | awk '/link/ {print $2}')
```

Comment:
The `dev` name `ens3f0` seems like environment specific can this be generalized?

Reply:
done